index.js

```diff
@@ -126,15 +144,27 @@ class ServerlessWebpack {
       'webpack:serve:serve': () => BbPromise.bind(this)
         .then(() => BbPromise.reject(new this.serverless.classes.Error('serve has been removed. Use serverless-offline instead.'))),
 
-      'before:offline:start': () => BbPromise.bind(this)
-        .then(this.prepareOfflineInvoke)
-        .then(this.compile)
-        .then(this.wpwatch),
+      'before:offline:start': () => {
+        if (this.options.functionObj.webpack === false) {
```

Comment:
Offline is used to simulate the service, not a single function (and will recompile in case of a file changed).
So imo the checks on `offline:start` and `offline:start:init` must be removed, as this would be never called.